### ISSUE TITLE
cli implementation uses "real" keywords commands

### ISSUE THREAD

**Issue by ivan1993br** [rock-core/autoproj-ci#7]
Hi all,

The implementation of functions that use keywords arguments uses the new feature of "real" keyword arguments, that was implemented on ruby version >= 2.5. For example on function:
```ruby
cli.create_report(*args, **options)
```
Will raise an errors on ruby versions <2.5, like the following:
```ruby
/home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/bundler/gems/autoproj-ci-5699720f6754/lib/autoproj/cli/ci.rb:176:in `create_report': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/bundler/gems/autoproj-ci-5699720f6754/lib/autoproj/cli/main_ci.rb:167:in `create_report'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /home/buildbot/.local/share/autoproj/gems/ruby/2.4.0/bundler/gems/autoproj-385aa63fadb7/bin/autoproj:8:in `<top (required)>'
	from /home/buildbot/dev/.autoproj/bin/autoproj:45:in `load'
	from /home/buildbot/dev/.autoproj/bin/autoproj:45:in `<main>'

```
As **discussed** [here](https://bugs.ruby-lang.org/issues/10708), and **implemented** [here](https://bugs.ruby-lang.org/issues/14183)

My questions are: Is this intended? Do you intend to port this to ruby versions <2.5 (native on ubuntu xenial for example)?

Thanks,

Ivan

**Comment by doudou** [comment]
It's more complicated than just "using real keyword arguments". Ruby has subtly changed behavior of keyword arguments (which were introduced in Ruby 2.0) in some corner cases, and that's what is being hit here.

Now, on compatibility: the Rock policy is to guarantee compatibility with the current Ubuntu LTS. 18.04 shipped (two years ago) with Ruby 2.5.

Even though I am trying to keep compatibility for older Ruby versions in Autoproj itself, this is a new plugin - as yet unreleased - I was definitely targeting 2.5.

Now, if **you** do want to step up and help maintaining pre-2.5 compatibility, I think we can work together on this. I just won't personally spend a lot of time testing on old ruby versions.

**Comment by doudou** [comment]
Hi @ivan1993br. Haven't heard back from you, so I'm closing this for now. Feel free to contribute a PR that makes this autoproj plugin compatible with 2.4 or 2.3, as long as it's not too intrusive.
